### PR TITLE
Setup Ruby Test CI actions

### LINKED ISSUE #23: Set up CI for tests
Reference from https://github.com/girijaasoni/newsletterApplicaton/pull/21

https://github.com/girijaasoni/newsletterApplicaton/pull/21#discussion_r1048407488
https://github.com/girijaasoni/newsletterApplicaton/pull/21#discussion_r1053226050
https://github.com/girijaasoni/newsletterApplicaton/pull/21#discussion_r1053226453

### PR DESCRIPTION
**Pull request by archanaserver**
Resolves #23 Set up CI for Ruby tests

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
I think we need to change the `database.yml.example` structure,
locally this is working for me:
```

default: &default
  adapter: postgresql
  encoding: unicode
  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>

development:
  <<: *default
  database: newsletterApplicaton_development
  port: 5432

test:
  <<: *default
  database: newsletterApplicaton_test
  port: 5432

production:
  <<: *default
  database: newsletterApplicaton_production
  username: newsletterApplicaton
  password: <%= ENV["NEWSLETTERAPPLICATON_DATABASE_PASSWORD"] %>
```

IMHO we can drop the `host, username, password` fields, what do you think @archanaserver @girijaasoni ?

**Comment by archanaserver** [comment]
@Ron-Lavi yeah, it is workimg for me too in locally. I'm looking into it, why it's failing here.

File: .github/workflows/ruby-ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
```
Comment (Ron-Lavi) [inline .github/workflows/ruby-ci.yml]: I think the env variables should be on the top of the file. e.g:

```suggestion
name: Ruby Testing
on: [pull_request]
env:
  RAILS_ENV: test
  DATABASE_URL: postgresql://postgres:@localhost/test
  DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
jobs:
  test:
    runs-on: ubuntu-latest
    services:
        postgres:
            image: postgres:12.1
            ports: ['5432:5432']
            options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
```

File: config/database.yml.example [hunk 2]
```diff
@@ -1,8 +1,11 @@
-
 default: &default
   adapter: postgresql
+  host: localhost
   encoding: unicode
+  username: <%= ENV["POSTGRES_USER"] %>
+  password: <%= ENV["POSTGRES_PASSWORD"] %>
```
Comment (Ron-Lavi) [inline config/database.yml.example]: can you also drop the host, username and password?

**Review by Ron-Lavi — APPROVED** [review]
Thanks @archanaserver, everything is green :)

**Pull request merged**